lib/merkDb/MerkDbTransactionWrapper.js

```diff
@@ -0,0 +1,97 @@
+class MerkDbTransactionWrapper {
+  /**
+   *
+   * @param {Merk} merkDB
+   */
+  constructor(merkDB) {
+    this.db = merkDB;
+    this.deleted = new Set();
+    this.data = new Map();
+  }
+
+  /**
+   *
+   * @param {Buffer} key
+   * @return {null|Buffer}
+   */
+  get(key) {
+    if (this.deleted[key.toString('hex')]) {
+      return null;
+    }
+
+    const value = this.data.get(key.toString('hex'));
+    if (value !== undefined) {
+      return value;
+    }
+
+    return this.db.getSync(key);
+  }
+
+  /**
+   *
+   * @param {Buffer} key
+   * @param {*} value
+   *
+   * @return {MerkDbTransactionWrapper}
+   */
+  put(key, value) {
+    this.deleted.delete(key.toString('hex'));
+
+    this.data.set(key.toString('hex'), value);
+
+    return this;
+  }
+
+  /**
+   *
+   * @param {Buffer} key
+   *
+   * @return {MerkDbTransactionWrapper}
+   */
+  delete(key) {
+    this.data.delete(key.toString('hex'));
```

Comment:
I would try to get (`this.getSync`) it first and then delete it if present just to mimic to merkDb behaviour.